### ISSUE TITLE
HLS ConcatenaningMediaSource does not autoplay when seeking to the end of video

### ISSUE THREAD

**Issue by q-litzler** [google/ExoPlayer#2736]
### Reproduction steps

I'm using the bare ExoPlayer, not the SimpleExoPlayer.

In 2.3.1:

When applying a track selection override to a HLS playlist (ConcatenatingMediaSource) and seeking manually to the very end of a video, the player is stuck in the buffering state, and the next video is not played automatically how it should be. However, if I seek to the end of the video minus a few second, and let the playback reach the end on his own, it does autoplay properly.

In 2.4.0:

Same thing, except you don't even need to apply the track selection override to reproduce the behaviour: Just seek to the very end of a video in the playlist, and it will remain stuck.

Plus, I'd like to say that there is no meaningful example or sample of how to use the DefaultTrackSelector. The track selection code in the demo app is a mess, and there is no real doc on how to use it.

### Link to test content
```
 val assets = listOf(
            "http://sample.vodobox.net/skate_phantom_flex_4k/skate_phantom_flex_4k.m3u8",
            "http://playertest.longtailvideo.com/adaptive/oceans_aes/oceans_aes.m3u8",
            "http://content.jwplatform.com/manifests/vM7nH0Kl.m3u8"
).map { HlsMediaSource(Uri.parse(it), player.dataSourceFactory, player.handler, null) }
val playlist = ConcatenatingMediaSource(*assets.toTypedArray())
```
I did not try with other types of playlist

### Version of ExoPlayer being used
2.3.1 and 2.4.0

### Device(s) and version(s) of Android being used
Google Pixel 7.1.2 and Nexus 6P 7.1.2

### A full bug report captured from the device
nope

**Comment by AquilesCanta** [comment]
I don't have much time to look into this right now so it will have to wait until I have some time or a teammate picks it up.

In the latter case, this seem only to reproduce when a text track is enabled in the currently played source. So, demo app reproduction steps:

1. Create a playlist, like so:

```
{
        "name": "HLS PLAYLIST",
        "playlist": [
          {
            "uri": "https://devimages.apple.com.edgekey.net/streaming/examples/bipbop_4x3/bipbop_4x3_variant.m3u8"
          },
          {
            "uri": "https://devimages.apple.com.edgekey.net/streaming/examples/bipbop_16x9/bipbop_16x9_variant.m3u8"
          }
        ]
      },
```

2. Play using the demo app. Make sure the text track is enabled. Might not be by default.
3. Seek to the very end of the seek bar.

Expected behavior: Playback should transition to the second media source.
Actual behavior: The player goes into an endless buffering state.

**Comment by ojw28** [comment]
It's possible that this is the same as #1874, although let's leave them as separate issues until we have a chance to confirm. It's relatively likely we'll get around to fixing #1874 first, at which point we should probably just check this issue again and see if it still reproduces.

**Comment by q-litzler** [comment]
Thank you for the prompt replies, and happy bug hunting ! :)

**Comment by ojw28** [comment]
#1874 is fixed in `dev-v2`. Would it be possible for you to see whether this issue is fixed for you as well in that branch?

**Comment by ojw28** [comment]
Closing based on the assumption this is a duplicate of #1874 or #1897, both of which are fixed in the `dev-v2` branch and will be included in the `2.6.0` release.
